### ISSUE TITLE
Incorrect Layer Normalization description

### ISSUE THREAD

**Issue by lamazavr** [tensorflow/tensorflow#39942]
Hi,
I see a [description of LayerNormalization](https://www.tensorflow.org/addons/tutorials/layers_normalizations).
According to this image:
![image](https://user-images.githubusercontent.com/1153259/83129373-64f5b280-a0e5-11ea-9f66-8f4ae4e3fbb2.png)
LayerNorm should compute mean and var across dimensions (C, H, W), thus I expect them to be 1D tensors of length N.
But the code of [tf.keras.layers.LayerNormalization](https://github.com/tensorflow/tensorflow/blob/2b96f3662bd776e277f86997659e61046b56c315/tensorflow/python/keras/layers/normalization.py#L1131) computes mean and var on 3rd dimension (which is C) and generates tensors of shape (N,H,W,1)

That seems incorrect, am I missing something?

**Comment by MoustafaMeshry** [comment]
I believe this is a bug with the implementation of LayerNormalization! 
A brief sanity check shows that the current implementation of tf.keras.layers.LayerNormalization() is actually an implementation of the instance normalization layer (e.g. tfa.layers.InstanceNormalization)
```
import tensorflow as tf
import tensorflow_addons as tfa

x = tf.zeros((4, 16, 16, 3))
layer_norm = tf.keras.layers.LayerNormalization()
inst_norm = tfa.layers.InstanceNormalization()
y1 = layer_norm(x)
y2 = inst_norm(x)
print(layer_norm.trainable_variables)  # expects gamma and beta variables to be scalars
print(inst_norm.trainable_variables)    # expects gamma and beta variables to be vectors of length = number of channels (i.e 3)
```

The output shows that both layers have the same number of trainable weights, which is incorrect for the Layer normalization.

**Comment by lamazavr** [comment]
Seems that tf.keras.layers.LayerNormalization corresponds with the InstanceNormalization image as well. So I think @MoustafaMeshry is right. There is a bug in tf.keras.layers.LayerNormalization.
@yhliang2018 what do you think?

**Comment by RomainBrault** [comment]
This was also reported in the tensorflow_addons repository. I am also convinced that there is a problem with Tensorflow' s `LayerNormalization()` and the correct behavious is `GroupNormalization(groups=1)`

https://github.com/tensorflow/addons/issues/1720

**Comment by lingyixia** [comment]
I suppose the default "axis"  param in LayerNormalization should be [1:] rather than -1

**Comment by lingyixia** [comment]
I suppose the default "axis" param in LayerNormalization should be [1:] rather than -1 which confuse me a lote

**Comment by RomainBrault** [comment]
On master there is now a comment explaining this behavious:

```
  Note that other implementations of layer normalization may choose to define
  `gamma` and `beta` over a separate set of axes from the axes being
  normalized across. For example, Group Normalization
  ([Wu et al. 2018](https://arxiv.org/abs/1803.08494)) with group size of 1
  corresponds to a Layer Normalization that normalizes across height, width,
  and channel and has `gamma` and `beta` span only the channel dimension.
  So, this Layer Normalization implementation will not match a Group
  Normalization layer with group size set to 1.
```

I agree that this is confusing, and the reasons why they don't follow the paper implementation proposal are not stated.

**Comment by Saduf2019** [comment]
@dsalnikov 
Could you please let us know if this is still an issue, can you verify with later versions of tf and let us know.

**Comment by MoustafaMeshry** [comment]
@Saduf2019 Yes, this is still an issue. As @lingyixia mentioned, the default value for the 'axis' argument should correspond to [1:len(input_shape)] (i.e all channel dimensions except for the batch dimension). Since the init() function does not have access to the input_shape, the fix probably requires a few more than a single line change.

**Comment by dsalnikov** [comment]
@Saduf2019, yes this is still an issue. Implementation doesn't match the document.

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
